### ISSUE TITLE
Vendors are not refreshed without logout/login

### ISSUE THREAD

**Issue by tomasfejfar** [keboola/developer-portal-ui#399]
* register as new user
* ask to be added to `keboola` vendor
* add user to vendor manually, not via accepting the request (this could be the issue)

Expected:
* when user refreshes the screen saying "You're not a member of any vendor." new vendor is shown

Actual:
* user needs to logout and login again to see newly added vendor

**Comment by ujovlado** [comment]
I understand this is not expected behavior, but vendors membership is stored within token - so login/logout is needed. End user (UI) doesn't know about that change.

This can be fixed in future using some websockets, etc. and changes will be pushed to client.

**Comment by JakubMatejka** [comment]
To sice je v tokenu, ale tohle byl jeden z důvodů proč jsem od toho od začátku zrazoval. Aktuální info o vendorech se dá získat z profilu uživatele: https://kebooladeveloperportal.docs.apiary.io/#reference/0/auth/get-user-profile

**Comment by ujovlado** [comment]
Tu ale ide skor o to ako tu informaciu dostat po zmene na klienta, nie odkial ju zobrat. Je mi v podstate jedno, ci spravim refresh tokenu alebo call na get profile. (Aj ked stale povazujem to, ze je to v tokene za prijatelnejsie - ale to je zase ina debata)

**Comment by tomasfejfar** [comment]
Tak bys mohl zapnout nějaký periodický check na refresh tokenu v případě, že uživatel provedl nějakou akci, která to vyžaduje. Nebo provést refresh po načtení stránky (tj. když udělá F5, tak dostane nový).

**Comment by ujovlado** [comment]
Neviem, no. Pre mna je to "super minor" vec. Token sa ti stejne refreshne do hodiny, a potom sa tie info o vendoroch dotiahnu. Takisto sa robi refresh tokenu, tak ako pises, pri akciach, kde je to potrebne - pridal si sa do vendora, odobral si sa z vendora a pod. (ale len vzhladom na aktualne usera, co robi v UI).

Nie je tam teda osetrena situacia, ktoru popisujes, ze mi niekto upravi clenstvo vo vendorovi. Vtedy, ako pisem, sa to prejavi do hodiny.

**Comment by ujovlado** [comment]
Ten periodicky check by tomu vlastne len zmenil "timeout", co sa mi nepaci - mame to schvalne tak, ze token sa refreshne az ked je to potrebne.

Po refreshi (f5) by to ale myslim slo - to by som mohol dorobit.

**Comment by tomasfejfar** [comment]
Neříkám, že je to nějak kritické. Ale je to nepříjemná věc v té kritické fázi onboardingu. Někdo zkouší portal, nechá se přidat do vendoru. A jako já pinguje "tak mě tam přijdejte". A oni "už tam jsi". Aha mě to píše "no vendors". "Nj, mají to nějaké pokopané". 

Good enough by bylo i kdyby tam byla message, že pro obnovení informací o vendorovi je potřeba se přihlásit nebo odhlásit. Případně počkat hodinu. Je to spíš o expectation management. Podobně jako s tím 5minutovým timeoutem než se přenese tag do KBC. Je to pochopitelné. Dá se s tím počítat. Ale musím to vědět.

**Comment by ujovlado** [comment]
Jo, chapem. Tam hodne zalezi, ci ta niekto pozyva alebo sa pridavas sam (davas req. na join do vendora) a ci ti to schvalia hned alebo nie.

Nieco s tym vymyslime :). Kym to bude, kde by podla teba mohla byt ta message?

**Comment by tomasfejfar** [comment]
Podle mého by se message mohla zobrazit jako flashmessage po tom, co si člověk vyžádá přidání k vendorovi. "Your assigned vendors are stored locally, so you may need to logout and login or wait up to 60 minutes for the information to propagate after you're approved."
